keras/layers/variational.py

```diff
@@ -0,0 +1,104 @@
+from keras import backend as K
+from keras.layers.core import Layer
+from keras import initializations, activations
+
+from keras.regularizers import GaussianKL
+
+
+class VariationalDense(Layer):
+    """VariationalDense
+        Hidden layer for Variational Autoencoding Bayes method [1].
+        This layer projects the input twice to calculate the mean and variance
+        of a Gaussian distribution. During training, the output is sampled from
+        that distribution as mean + random_noise * variance, during testing the
+        output is the mean, i.e the expected value of the encoded distribution.
+
+        Parameters:
+        -----------
```

Comment:
With the introduction of autogenerated documentation in December, the docstrings are use markdown. You can check out existing layers for examples.